### ISSUE TITLE
TOML parser crashes for an invalid content

### ISSUE THREAD

**Issue by sameerajayasoma** [ballerina-platform/ballerina-lang#28400]
Here is the content:

```toml

log.a = 10
log.b = "sameera"

[log.a]
a = 15


```

error:
```
[2021-02-02 16:37:20,622] SEVERE {b7a.log.crash} - class io.ballerina.toml.semantic.ast.TomlTableNode cannot be cast to class io.ballerina.toml.semantic.ast.TomlKeyValueNode (io.ballerina.toml.semantic.ast.TomlTableNode and io.ballerina.toml.semantic.ast.TomlKeyValueNode are in unnamed module of loader 'app')
java.lang.ClassCastException: class io.ballerina.toml.semantic.ast.TomlTableNode cannot be cast to class io.ballerina.toml.semantic.ast.TomlKeyValueNode (io.ballerina.toml.semantic.ast.TomlTableNode and io.ballerina.toml.semantic.ast.TomlKeyValueNode are in unnamed module of loader 'app')
	at io.ballerina.runtime.internal.configurable.ConfigTomlParser.validateNodeAndExtractValue(ConfigTomlParser.java:109)
	at io.ballerina.runtime.internal.configurable.ConfigTomlParser.populateConfigMap(ConfigTomlParser.java:90)
	at io.ballerina.runtime.internal.launch.LaunchUtils.initConfigurableVariables(LaunchUtils.java:134)
	at sameera.log.0_1_0.$ConfigurationMapper.$configureInit(Unknown Source)
	at sameera.log.0_1_0.$_init.main(log)
```

**Comment by sameerajayasoma** [comment]
This may not be an issue in the parser itself. jBallerina runtime configurable impl causes this.

**Comment by HindujaB** [comment]
This happens due to TOML parser returning unexpected value nodes for the dotted keys. 
https://github.com/ballerina-platform/ballerina-lang/issues/27789

Though, the error needs to be handled without giving a bad sad error.

**Comment by xlight05** [comment]
This issue was fixed with  PR #28329 . Still, there was the issue with duplicate error messages for this scenario. #28487 will fix that as well. :)
